### PR TITLE
112 - video reducer unit tests (and a few other things)

### LINKED ISSUE #112: Add some reducer unit tests
Let's start with `VideoReducer.js`

### PR DESCRIPTION
**Pull request by bent0b0x**
**Work Done**
- add unit tests for all actions handled by the videos reducer
- remove a reducer path that wasn't being used
- change/fix some logic in the reducers (with a focus on ensuring edge cases are covered and that the whole state remains immutable)
- add a scripts/configure circle so that chimp tests do not start until dev server is responsive

@jellegerbrandy I changed code in all of the video reducer code paths, so feel free to do some testing to make sure nothing is screwed up. I do not know the uploader as well at this point to be able to tell for sure that things are working 100% as designed. (Also I think the transcoder is still not working so difficult to test the entire workflow).

Resolves #112, #160, and #166

### CODE REVIEW

File: src/scripts/components/VideoForm.js [hunk 1]
```diff
@@ -117,36 +117,51 @@ class VideoForm extends Component<Props, Object> {
   }
 
   componentWillReceiveProps (nextProps: Props): void {
-    this.setState(nextProps.selectedVideo)
-    this.setState({
-      id: nextProps.selectedVideo.id,
-      title: nextProps.selectedVideo.title,
-      description: nextProps.selectedVideo.description
-    })
+    const { selectedVideo } = nextProps
```
Comment (bent0b0x) [inline src/scripts/components/VideoForm.js]: The changes in non-reducer files were mainly bugs that were uncovered during work on this PR. Mostly that some props could be `undefined/null` in components and that case was not accounted for in the components themselves.

File: src/scripts/waitForServer.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        process.exit(1)
+      }
+    }
+  )
+
+  req.on('error', pingServer)
+  req.end()
+}
+
+pingServer()
```
Comment (bent0b0x) [inline src/scripts/waitForServer.js]: Here's the hacky little util I wrote to not start our chimp tests until the server responds. So far, it seems to be working!

File: test/unit-tests/reducers/VideosReducer.test.js [hunk 3]
```diff
@@ -0,0 +1,1588 @@
+/* @flow */
```
Comment (bent0b0x) [inline test/unit-tests/reducers/VideosReducer.test.js]: Enjoy reading through this diff...
